lib/Doctrine/Common/Collections/ArrayCollection.php

```diff
@@ -396,7 +409,9 @@ public function matching(Criteria $criteria)
                 $next = ClosureExpressionVisitor::sortByField($field, $ordering === Criteria::DESC ? -1 : 1, $next);
             }
 
-            uasort($filtered, $next);
+            if ($next) {
+                uasort($filtered, $next);
```

Comment:
I would like to actually see reason why was this change sneaked here

Reply:
According to the loop above, `$next` may be `null` from a type perspective. That is impossible to reach, assuming `ClosureExpressionVisitor::sortByField()` is not nullable

Reply:
I think Psalm should actually be able to figure that out, given that it's set in the loop over non-empty `$orderings`. I'm going to file Psalm bug for this.

Reply:
Ticketed under vimeo/psalm#1470.

Reply:
Thanks, I've fixed (but it'll be a few days yet before I release).